library/core/src/slice/iter.rs

```diff
@@ -49,7 +49,7 @@ impl<'a, T> IntoIterator for &'a mut [T] {
 /// // First, we need a slice to call the `iter` method on:
 /// let slice = &[1, 2, 3];
 ///
-/// // Then we call `iter` on the slice to get the `Iter` struct,
```

Comment:
This is actually correct as-is. Given that it's documentation on the `Iter` type itself, there shouldn't be any confusion here. Likewise throughout.

Reply:
This is certainly correct, but later prose talks about calling the X method "of the iterator", without an iterator having been mentioned before. If you think it is less clear without struct, perhaps we can use something like "the `Iter` struct (which is an iterator)" or "the `Iter` struct (which implements the Iterator trait)"? What do you think?